client/src/components/Header/Header.js

```diff
@@ -34,97 +34,83 @@ export default class Header extends React.Component {
-                                    </ul>
-                                </div>
-                            </CheeseburgerMenu>
+                    </div>
+                </div>
 
+                <div className="center">
+                    {/* Desktop and Tablet */}
+                    {window.innerWidth >= 576 && <>
+                        <div className="search-bar-container">
+                            <SearchBar/>
                         </div>
-                    </>
-                }
+                    </>}
+
+                    {/* Mobile */}
+                    {window.innerWidth < 768 && <>
+                        <Modal className="mobile-search-modal" show={this.state.isSearchModalOpen} onHide={() => {this.setState({isSearchModalOpen: false})}} animation={false}>
+                            <Modal.Header closeButton>
+                                <img className='header-icon' src={process.env.PUBLIC_URL + '/MM-logo-rev-3.png'} alt="Monuments and Memorials Logo" width='35px' height='35px'/>
+                            </Modal.Header>
+                            <Modal.Body>
+                                <SearchBar onCloseModal={() => {
+                                    this.setState({isSearchModalOpen: false})
+                                }}/>
+                            </Modal.Body>
+                        </Modal>
+                    </>}
+                </div>
+
+                <div className="right">
+                    {/* Desktop */}
+                    {window.innerWidth >= 992 && <>
+                        <Button size="sm" variant="link-secondary">Log in</Button>
+                        <Button size="sm">Sign up</Button>
+                    </>}
+                    {/* Mobile */}
+                    {window.innerWidth < 768 && <>
+                        <Button variant="link-secondary" className="search-icon" onClick={() => this.setState({isSearchModalOpen: true})}>
+                            <i className="material-icons">
+                                search
+                            </i>
+                        </Button>
+                    </>}
+                    {/* Tablet and Mobile */}
+                    {window.innerWidth < 992 && <>
+                        <Button variant="link-secondary" className="menu-icon" onClick={() => this.setState({isMenuOpen: true})}>
+                            <i className="material-icons">
+                                menu
+                            </i>
+                        </Button>
+                        <CheeseburgerMenu isOpen={this.state.isMenuOpen} closeCallback={() => {this.setState({isMenuOpen: false})}}>
+                            <ul>
+                                <div className="links d-lg-block">
+                                    {headerLinks.map(link =>
+                                        <li>{link}</li>
+                                    )}
```

Comment:
Rather than copy and paste the link code, I did it once in a variable at the start of the render function and then here I just wrap each link in a `<li>` so that it works in the `<ul>`